tests/network/checkup_framework/utils.py

```diff
@@ -256,13 +256,15 @@ def wait_for_job_failure(job):
         job_status = TimeoutSampler(
             wait_timeout=timeout,
             sleep=1,
-            func=lambda: job.instance.status.conditions[0],
+            func=lambda: job.instance.status.conditions,
         )
         for sample in job_status:
-            if sample["type"] == job.Status.FAILED and sample["status"] == job.Condition.Status.TRUE:
-                return
-            if sample["type"] == job.Status.SUCCEEDED and sample["status"] == job.Condition.Status.TRUE:
-                raise ResourceValueError(f"Job {job.name} has succeeded and should have failed.")
+            if sample:
+                for condition in sample:
+                    if condition["type"] == job.Status.FAILED and condition["status"] == job.Condition.Status.TRUE:
+                        return
+                    if condition["type"] == job.Status.SUCCEEDED and condition["status"] == job.Condition.Status.TRUE:
+                        raise ResourceValueError(f"Job {job.name} has succeeded and should have failed.")
```

Comment:
you might want to use `@retry` decorator instead of timeout sampler this way.

Reply:
Interesting.
Maybe in another PR.